packages/sirius-web/backend/sirius-web-domain/src/main/java/org/eclipse/sirius/web/domain/boundedcontexts/library/Library.java

```diff
@@ -0,0 +1,254 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.web.domain.boundedcontexts.library;
+
+import java.time.Instant;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.Objects;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import org.eclipse.sirius.components.events.ICause;
+import org.eclipse.sirius.web.domain.boundedcontexts.AbstractValidatingAggregateRoot;
+import org.eclipse.sirius.web.domain.boundedcontexts.library.events.LibraryContentUpdatedEvent;
+import org.eclipse.sirius.web.domain.boundedcontexts.library.events.LibraryCreatedEvent;
+import org.eclipse.sirius.web.domain.boundedcontexts.library.events.LibraryDeletedEvent;
+import org.eclipse.sirius.web.domain.boundedcontexts.library.events.LibraryDependencyAddedEvent;
+import org.eclipse.sirius.web.domain.boundedcontexts.library.events.LibraryDependencyRemovedEvent;
+import org.eclipse.sirius.web.domain.boundedcontexts.library.events.LibraryDescriptionUpdatedEvent;
+import org.eclipse.sirius.web.domain.boundedcontexts.library.events.LibraryNameUpdatedEvent;
+import org.eclipse.sirius.web.domain.boundedcontexts.library.events.LibraryVersionUpdatedEvent;
+import org.eclipse.sirius.web.domain.boundedcontexts.project.Project;
+import org.springframework.data.annotation.Id;
+import org.springframework.data.annotation.Transient;
+import org.springframework.data.domain.Persistable;
+import org.springframework.data.jdbc.core.mapping.AggregateReference;
+import org.springframework.data.relational.core.mapping.Column;
+import org.springframework.data.relational.core.mapping.MappedCollection;
+import org.springframework.data.relational.core.mapping.Table;
+
+/**
+ * The aggregate root of the library bounded context.
+ *
+ * @author gdaniel
+ */
+@Table("library")
+public class Library extends AbstractValidatingAggregateRoot<Library> implements Persistable<UUID> {
```

Comment:
I don't understand why a single part (apart maybe the description and even there I'm not entirely convinced) of a library can be updated. Libraries should be IMMUTABLE, one should not be able to change the name, content, dependencies or anything in a library.

One cannot update a maven package once published. It's done, forever and for very good reasons.

Reply:
All the events can be removed too